components/service/pocket/src/main/java/mozilla/components/service/pocket/stories/update/RefreshPocketWorker.kt

```diff
@@ -0,0 +1,73 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.pocket.stories.update
+
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import androidx.work.CoroutineWorker
+import androidx.work.Data
+import androidx.work.WorkerParameters
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import mozilla.components.service.pocket.DEFAULT_ARTICLES_COUNT
+import mozilla.components.service.pocket.R
+import mozilla.components.service.pocket.stories.PocketStoriesUseCases
+
+/**
+ * WorkManager Worker used for downloading and persisting locally a new list of Pocket recommended articles.
+ */
+internal class RefreshPocketWorker(
+    private val context: Context,
+    params: WorkerParameters
+) : CoroutineWorker(context, params) {
+
+    override suspend fun doWork(): Result {
+
+        val pocketToken = inputData.getString(KEY_EXTRA_DATA_POCKET_TOKEN)
+        val pocketItemsLocale = inputData.getString(KEY_EXTRA_DATA_POCKET_ITEMS_LOCALE)
+        val pocketItemsCount = inputData.getInt(KEY_EXTRA_DATA_POCKET_ITEMS_COUNT,
+            DEFAULT_ARTICLES_COUNT
+        )
+
+        return withContext(Dispatchers.IO) {
+            if (PocketStoriesUseCases().RefreshPocketStories(
+                    context, pocketToken!!, pocketItemsCount, pocketItemsLocale!!
+                ).invoke()
+            ) {
+                Result.success()
+            } else {
+                Result.failure()
```

Comment:
@Amejia481 and I ran into this before and found that returning `failure` here actually prevents the periodic task from running again, which is not what you want in this case. We'd have to verify again, but then you'd want to return `retry` or `success` here instead.

Reply:
Yeah,  you can find some docs [here](https://developer.android.com/reference/androidx/work/ListenableWorker.Result#failure()), additionally we did something for `AddonUpdaterWorker` where [we stopped a periodic work](https://github.com/mozilla-mobile/android-components/blob/a9e5357a4ea19ceb07de96b3ccf5e72a56d3b427/components/feature/addons/src/main/java/mozilla/components/feature/addons/update/AddonUpdater.kt#L591) from getting executed when add-on was not installed anymore.

Reply:
Thank you!